src/utp_wrapper.rs

```diff
@@ -47,7 +49,13 @@ impl UtpWrapper {
                                     }
                                 },
                                 Err(TryRecvError::Disconnected) => break 'outer,
-                                Err(TryRecvError::Empty) => break,
+                                Err(TryRecvError::Empty) => {
+                                    if timeout < ::time::SteadyTime::now() {
```

Comment:
the outer loop (the socket connection thread) needs to be broken (existed) once there is continuously timed-out events for a long period.
this is the main point causing the hanging issue on windows as the dangling thread will keeps waiting forever.